### DISCUSSION TITLE
Welcome to LiLo.Lite Discussions!

### DISCUSSION THREAD

**Discussion by GeorgeLeithead** [GeorgeLeithead/LiLo.Lite#10] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by yekcoin** [comment]
Hi,
Thanks for your share code.

Lilo.Lite working properly on Android 9.0. 
But not working Android 8.0 <= (Such as Samsung Galaxy S7 Android 8.0 - API 26)
How can i fix?
Thanks.

**Reply by GeorgeLeithead** [reply]
Hi yekcoin,

I can see the BASIC error details in the AppCentre issues log.  I can see that you have cloned the repository, and that the error 'appears' to be from the initialisation of HomeView, and some "Modules\Register\RegisterViewModel.cs" that must be unique to your clone copy?  Do you know when you cloned a copy, as I have made a lot of updates, as recent as yesterday and have lots more coming...today?

I have tested the app on an emulated Google Nexus One, running Lollipop 5.0 API 21, and physically on a Samsung Galaxy S6 (also Android 8.0 - API 26) without issues.

I therefore expect its something to do with the clone that is trying to access something not set-up, or possibly any additions made (see note above).

From the AppCentre issues log, it unfortunately only includes that it crashes on the InitialiseComponent of the HomeView, which is not overly helpful.

FIRSTLY
=====
I suggest you clone THE LATEST version into a separate location and validate that it works without any modifications.  If it works, then that either points to I've already 'fixed' the issue, or there is an issue with any changes/additions you made.

SECONDLY
=======
Debugging in Xamarin.Forms is really not that helpful, but I do suggest that to find the issue that we use it.  As such I suggest that you run in debug, and attach breakpoints and then STEP thru each to narrow WHERE it's actually crashing (debug really IS bad for Xamarin.Forms).  I suggest:

- HomeView.xaml.cs - 1st line in the OnAppearing method
-    base.OnAppearing();
- App.xaml.cs - in the App method, when registering the SocketsService dependency
-    DependencyService.Register<SocketsService>();
- MarketsHelperService.cs
-    IEnumerable<MarketModel> markets = DataStore.GetMarketsForFeed();

It's then a case of stepping thru to find the TRUE location of the issue, and then identifying why.


Keep me informed as to how you get on.
